packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSource.ts

```diff
@@ -148,15 +215,17 @@ export const useGridDataSource = (
         groupKeys: rowNode.path,
       };
 
-      const cachedData = apiRef.current.unstable_dataSource.cache.get(fetchParams);
+      const cacheKeys = cacheChunkManager.getCacheKeys(fetchParams);
+      const responses = cacheKeys.map((cacheKey) => cache.get(cacheKey));
+      const cachedData = responses.some((response) => response === undefined)
+        ? undefined
+        : CacheChunkManager.mergeResponses(responses as GridGetRowsResponse[]);
 
       if (cachedData !== undefined) {
         const rows = cachedData.rows;
         nestedDataManager.setRequestSettled(id);
         apiRef.current.updateServerRows(rows, rowNode.path);
-        if (cachedData.rowCount) {
-          apiRef.current.setRowCount(cachedData.rowCount);
-        }
+        apiRef.current.setRowCount(cachedData.rowCount === undefined ? -1 : cachedData.rowCount);
```

Comment:
For consistency, would it make sense to handle nested response using strategy processing too?

Reply:
That can probably be handled in https://github.com/mui/mui-x/issues/14527